pkg/store/proxy.go

```diff
@@ -348,6 +343,21 @@ type streamSeriesSet struct {
 	closeSeries     context.CancelFunc
 }
 
+type recvResponse struct {
+	r   *storepb.SeriesResponse
+	err error
+}
+
+func startFrameCtx(responseTimeout time.Duration) (context.Context, context.CancelFunc) {
+	frameTimeoutCtx := context.Background()
```

Comment:
I would really suggest to actually get context from argument, (so from `startStreamSeriesSet` )

Reply:
Ignore me you already check the parent one, so all good